flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/decorators/FlinkConfMountDecorator.java

```diff
@@ -150,6 +173,24 @@ private Pod decoratePod(Pod pod) {
         return Collections.singletonList(flinkConfConfigMap);
     }
 
+    private Optional<File> getTaskManagerPodTemplateFile() {
+        return kubernetesComponentConf
+                .getFlinkConfiguration()
+                .getOptional(KubernetesConfigOptions.TASK_MANAGER_POD_TEMPLATE)
+                .map(
+                        file -> {
+                            final File podTemplateFile = new File(file);
+                            if (!podTemplateFile.exists()) {
+                                throw new FlinkRuntimeException(
+                                        new FileNotFoundException(
+                                                String.format(
+                                                        "Pod template file %s does not exist.",
+                                                        file)));
+                            }
+                            return podTemplateFile;
+                        });
```

Comment:
I think this breaks currently. If the user starts a cluster with `kubernetes.pod-template-file` pointing to a local file, then the started JobManager pod won't have this file but still tries to access it.